### PR TITLE
Allows group names for .obj and .lib. Parses directories for possible…

### LINKED ISSUE #140: Project - scrape_dir - additional arguments
The current scape dir does all the logic, scans and finds all valid files to form a yaml file. I would like to change this to split this into 2.
- scan function which can accept extension list and match keywords list, returns a dictionary
- create_yaml - instead of scrape_dir, will invoke scan function with all required data

It should allow us to scan a directory, return dictionary which some external tools can edit and then build/export via python api.

### LINKED ISSUE #146: Export function needs .yaml files in project section
Would like to have the option to replace .yaml files with the attributes that they eventually reference 

Example:

projects:
    kl25z_blinky:
        - records/kl25z_cmsis.yaml
        - records/kl25z_target.yaml
        - records/kl25z_blinky.yaml

Instead of needing to open  - records/kl25z_blinky.yaml

Should be able to replace that with contents of  -records/kl25z_blinky.yaml

### LINKED ISSUE #155: Obj or lib sources
I have consolidated sources recently for v0.6, and now thinking what we can do with

- source_files_obj
- source_files_lib

We probably can't make all into sources, as some libs might be without extension (although do we support  that?). I propose 3 solutions

1. All in sources - object files, libraries, source files (requires to fill proper extensions for us to group internally)
2. source_files_obj rename to object_files, source_files_lib to static_libraries
3. 1 and 2 combined - to resolve issue in the 1st one - using the second if needed. This would mean to keep both formats possible, which might be confusing. I would use the first one, and the second one would be used internally - once we sort sources to groups (source cpp, c, obj, libraries), we assing them to the proper keys - object_files and so on..

I vote for 3, with strong preference to use all in sources :-)

### PR DESCRIPTION
**Pull request by sarahmarshy**
… relative include paths in header files. This fixes #155, fixes #146, fixes #140

### CODE REVIEW

File: project_generator/templates/uvision4.uvproj.tmpl [hunk 1]
```diff
@@ -377,8 +377,7 @@
             <ScatterFile>{{linker_file}}</ScatterFile>
             <IncludeLibs></IncludeLibs>
             <IncludeLibsPath></IncludeLibsPath>
-            <Misc>{% for lib in source_files_lib %} {{lib}} {% endfor %}
-                  {% for obj in source_files_obj %} {{obj}} {% endfor %}</Misc>
+            <Misc></Misc>
```
Comment (0xc0170) [inline project_generator/templates/uvision4.uvproj.tmpl]: Where are now object  files and libraries? Part of the project?

Comment (sg-) [inline project_generator/templates/uvision4.uvproj.tmpl]: yes - they're in the project tree and with .ar and .a and .lib you can use the dialogs to choose which object files are used in the archive :)

Comment (0xc0170) [inline project_generator/templates/uvision4.uvproj.tmpl]: ok, was not aware of that, will test

File: project_generator/exporters/uvision.py [hunk 2]
```diff
@@ -146,12 +147,16 @@ def fix_paths(self, data, rel_path):
             fixed_paths.append(join(rel_path, normpath(path)))
         data['includes'] = fixed_paths
         fixed_paths = []
-        for path in data['source_files_lib']:
-            fixed_paths.append(join(rel_path, normpath(path)))
-        data['source_files_lib'] = fixed_paths
-        fixed_paths = []
-        for path in data['source_files_obj']:
-            fixed_paths.append(join(rel_path, normpath(path)))
+        for k in data['source_files_lib'][0].keys():
```
Comment (0xc0170) [inline project_generator/exporters/uvision.py]: this also expects as sources a dictionary, which is not valid always. If a project is just flat (no groups), this generates attribute error.

**Pull request merged**